public/app/core/logs_model.ts

```diff
@@ -356,7 +361,7 @@ export function logSeriesToLogsModel(logSeries: DataFrame[]): LogsModel | undefi
   if (limits.length > 0) {
     meta.push({
       label: 'Limit',
-      value: `${limits[0].meta.limit} (${rows.length} returned)`,
+      value: `${limit || limits[0].meta.limit} (${rows.length} returned)`,
```

Comment:
If we set the limit in the query won't it come back in the `meta` of the dataframe? Seems like that would make sense here and we could use that instead of propagating the limit separately.

Reply:
Yep, we get it back in `meta` of the dataframe, but data can come back in chunks. Already did a small fix to that.